crates/biome_js_analyze/src/lint/nursery/use_consistent_response.rs

```diff
@@ -0,0 +1,371 @@
+        return None;
+    };
+
+    let first_arg_any = first_arg.as_any_js_expression()?;
+    let is_first_arg_empty = match first_arg_any {
+        AnyJsExpression::AnyJsLiteralExpression(
+            AnyJsLiteralExpression::JsNullLiteralExpression(_),
+        ) => true,
+        AnyJsExpression::JsIdentifierExpression(expr) => {
+            expr.name().map(|ident| ident.is_undefined()).ok()?
+        }
+        AnyJsExpression::AnyJsLiteralExpression(
+            AnyJsLiteralExpression::JsStringLiteralExpression(expr),
+        ) => expr.inner_string_text().ok()?.text().is_empty(),
+        _ => return None,
+    };
+
+    if !is_first_arg_empty {
+        return None;
+    }
+
+    let mut status: Option<u16> = None;
+    let mut location: Option<Box<str>> = None;
+
+    loop_object(
+        second_arg.as_any_js_expression()?,
+        &mut |name, property| match name {
+            "status" => {
+                let value = property.value().ok()?;
+                let number = value
+                    .as_any_js_literal_expression()?
+                    .as_js_number_literal_expression()?
+                    .as_number()? as u16;
+
+                if REDIRECT_STATUS_CODES.contains(&number) {
+                    status = Some(number);
+                }
+
+                Some(())
+            }
+            "headers" => {
+                let object = property.value().ok()?;
+
+                loop_object(&object, &mut |name, property| {
+                    if name != "location" {
+                        return None;
+                    }
+
+                    let value = get_property_string_literal_value(property).take_if(|v| {
+                        // In server runtimes (Node, Deno, etc.) relative URLs will throw an error.
+                        // https://github.com/denoland/deno/issues/20674
+                        v.starts_with("http")
+                    })?;
+
+                    location = Some(value.text().into());
+
+                    Some(())
+                })
+            }
+            _ => None,
+        },
+    )?;
+
+    Some(ResponseSimplification::Redirect((status?, location?)))
+}
+
+/// Extracts the argument from a `JSON.stringify` call.
+fn extract_json_stringify_arg(
+    model: &SemanticModel,
+    call_expr: &JsCallExpression,
+) -> Option<AnyJsCallArgument> {
+    let callee = call_expr.callee().ok()?;
+    let callee = callee.as_js_static_member_expression()?;
+    let object = callee.object().ok()?;
+    let (reference, object_name) = global_identifier(&object.omit_parentheses())?;
+
+    if object_name.text() != "JSON" {
+        return None;
+    }
+
+    if model.binding(&reference).is_some() {
+        return None;
+    }
+
+    let args = call_expr.arguments().ok()?.args();
+
+    // If arguments are empty or contain more than one argument
+    // we can't use shorthand.
+    if args.len() != 1 {
+        return None;
+    }
+
+    args.first()?.ok()
+}
+
+fn loop_object(
```

Comment:
`loop_object()` doesn't seem a very descriptive function name. Why are you looping? Are you returning something from the loop?

I think I'd call this `for_each_object_member()`.